app/models/sentence.php

```diff
@@ -1096,5 +1114,54 @@ public function editAudio($sentenceId, $hasaudio)
         $this->id = $sentenceId;
         return $this->saveField('hasaudio', $hasaudio);
     }
+
+    /**
+     * Edit the sentence.
+     *
+     * @param  int $id      Sentence id.
+     * @param  string $text New sentence text.
+     * @param  string $lang New sentnece lang.
+     *
+     * @return boolean
+     */
+    public function editSentence($id, $text, $lang)
+    {
+        $this->id = $id;
+
+        if ($this->hasAudio($id)) {
+            return false;
+        }
+
+        $hash = $this->makeHash($lang, $text);
+
+        if ($duplicate = $this->findByBinary($hash, 'hash')) {
+            return false;
+        }
+
+        $data['Sentence']['text'] = $text;
+        $data['Sentence']['hash'] = $hash;
+
+        if (!empty($lang)) {
+            $data['Sentence']['lang'] = $lang;
+        }
+        
+        return $this->save($data);
+    }
+
+    /**
+     * Return true if sentence has audio.
+     *
+     * @return boolean
+     */
+    public function hasAudio($id)
+    {
+        $sentence = $this->findById($id);
+
+        if ($sentence['Sentence']['hasaudio'] === 'yes') {
```

Comment:
Should be `if ($sentence['Sentence']['hasaudio'] !== 'no')`

The value `yes` is not a valid value for `hasaudio`. Possible values are documented here: https://github.com/Tatoeba/tatoeba2/blob/dev/docs/database/tables/sentences.sql#L34